sa/database.go

```diff
@@ -156,12 +160,173 @@ func NewDbMapFromConfig(config *mysql.Config, settings DbSettings) (*boulderDB.W
 	dbmap := &gorp.DbMap{Db: db, Dialect: dialect, TypeConverter: BoulderTypeConverter{}}
 
 	initTables(dbmap)
-
 	return &boulderDB.WrappedMap{DbMap: dbmap}, nil
 }
 
-// adjustMySQLConfig sets certain flags that we want on every connection.
-func adjustMySQLConfig(conf *mysql.Config) {
+var checkStringQuoteRE = regexp.MustCompile(`^'[0-9A-Za-z_\-=:]+'$`)
+var checkIntRE = regexp.MustCompile(`^\d+$`)
+var checkImproperIntRE = regexp.MustCompile(`^'\d+'$`)
+var checkFloatOrIntRE = regexp.MustCompile(`^?\d+(\.\d+)?$`)
```

Comment:
I don't think you want the question mark at the very beginning of this regex -- isn't that making the `^` optional?

Reply:
Good catch, thanks!